pkg/api/handlers/utils/containers.go

```diff
@@ -202,12 +204,35 @@ func waitRemoved(ctrWait containerWaitFn) (int32, error) {
 	return code, err
 }
 
-func waitNextExit(ctrWait containerWaitFn) (int32, error) {
-	_, err := ctrWait(define.ContainerStateRunning)
-	if err != nil {
+func waitNextExit(ctx context.Context, containerName string) (int32, error) {
+	runtime := ctx.Value("runtime").(*libpod.Runtime)
+	var containerEngine entities.ContainerEngine = &abi.ContainerEngine{Libpod: runtime}
```

Comment:
```suggestion
	containerEngine := abi.ContainerEngine{Libpod: runtime}
```

Reply:
I deliberately used explicit type so I just call interface method, not method specific to the impl struct.